avocado/utils/genio.py

```diff
@@ -174,3 +181,21 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def write_file_or_fail(filename, data):
+    """
+    Write to a file and raise exception on write failure
+
+    :param filename : Path to file
+    :type filename : str
+    :param data : Data to be written to file
+    :type data : str
+    :raises GenIOException : On write Failure
+    """
+    fd = os.open(filename, os.O_WRONLY)
+    try:
+        ret = os.write(fd, data)
+    except OSError as details:
+        raise GenIOException("The write to %s failed: %s" % (
+                             filename, details))
```

Comment:
Oh yeah, I keep forgetting about this 'nice' feature of OSError's implementation not showing you which file actually failed to write, making the error opaque.

About the name, I have not noticed, but indeed `GenIOError` seems better and more consistent than `GenIOException`.